### ISSUE TITLE
Configurator will not install

### ISSUE THREAD

**Issue by inputd** [home-assistant/hassio#582]
Home assistant version: 0.74.0
supervisor version: 115

Brand new Pi B, fresh install. Installed Add-ons (Duck-DNS, Mosquitto broker, node-red), and have done no other customization.

When trying to add Configurator add-on, the following error appears in the system log:

18-07-20 22:37:33 ERROR (SyncWorker_14) [hassio.docker.interface] Can't install homeassistant/aarch64-addon-configurator:0.4 -> 500 Server Error: Internal Server Error ("readlink /var/lib/docker/overlay2: invalid argument").

**Comment by pvizeli** [comment]
That is a host problem, we can not install the add-on while the host docker have a trouble. Do you run HassOS?

**Comment by inputd** [comment]
Yes - it was the newest at the time, HassOS Release 1 build 5 (Stable)

**Comment by inputd** [comment]
I've updated hassOS to 1.7, but still have the same issue. This is my only machine with the 64-bit install, and the only one with this issue. Not sure if I should just reformat the thing or if this is a larger issue.

**Comment by frenck** [comment]
@inputd Just to be sure, is this still an issue?

**Comment by frenck** [comment]
No response in a month from the original submitter. Assuming the issue solved.
Feel free to re-open when needed.

**Comment by jacob80** [comment]
I am experiencing this same exact issue when attempting to install configurator:

`18-12-06 04:10:54 ERROR (SyncWorker_0) [hassio.docker.interface] Can't install homeassistant/armhf-addon-configurator:2 -> 500 Server Error: Internal Server Error ("readlink /var/lib/docker/overlay2: invalid argument").`

**Comment by Roberto6969** [comment]
Also I have same issue (slightly different readlink location):

19-02-06 12:31:05 ERROR (SyncWorker_10) [hassio.docker.interface] Can't install homeassistant/armhf-addon-configurator:2 -> 500 Server Error: Internal Server Error ("readlink /mnt/data/docker/overlay2/l: invalid argument").

**Comment by Hepokatti** [comment]
Similar problem here too. Trying to update Hassio 0.87.1 -> 0.88.0b1

HassOS version is 2.8 running on Pi3+.

19-02-16 23:49:04 ERROR (SyncWorker_9) [hassio.docker.interface] Can't install homeassistant/raspberrypi3-homeassistant:0.88.0b1 -> 500 Server Error: Internal Server Error ("readlink /mnt/data/docker/overlay2/l: invalid argument").

Any idea how this could be fixed?

**Comment by seaser0** [comment]
Same issue on mv HassOS VM 0.88.2 but only with the Configurator plugin. 
Tried to install on the admin page but same error. 
Tried also to install ith by CLI: 
`hassio addons install core_configurator --log-level debug`

I get also an error message but with an error code "400 Bad request".

Has someone experience in troubleshooting such issues?

**Comment by thadd** [comment]
@frenck This issue is not resolved. See the other responses. The same error arises on attempting to upgrade from 0.88.2 to latest version.

**Comment by frenck** [comment]
Check, let me re-open the issue in that case.

**Comment by pvizeli** [comment]
This error is a defect Docker database

**Comment by thadd** [comment]
> This error is a defect Docker database

Is this something that can easily be fixed with just SSH access to the Hassio container?

**Comment by gfetters** [comment]
I am getting this error after upgrading to latest version of hassos and configurator today.  I removed configurator to try and fix this and now I can't install configurator

**Comment by thadd** [comment]
@inputd I just did this and got the same `readline` error. Did you do a full or partial backup?

**Comment by inputd** [comment]
Full backup, and selected "wipe & restore".

You could also try doing a full backup sans Configurator, wipe & restore, and manually re-install configurator

**Comment by thadd** [comment]
> 

> Full backup, and selected "wipe & restore".
> 
> You could also try doing a full backup sans Configurator, wipe & restore, and manually re-install configurator

@inputd My current install doesn't include configurator since I tried removing it in an attempt to debug the inability to update and now can't install it anymore (same error as updating). Sadly I don't think the wipe & restore solution works for everyone...

I guess if it gets far enough out of date I'll just have to do a full reinstall from scratch and try to restore onto a fresh system. Very frustrating.

**Comment by thadd** [comment]
I finally gave in and did a full reinstall. I did the following:

* Created a snapshot via the web interface
* Copied the snapshot from the `/backup` directory to my local computer
* Followed the instructions [here](https://www.home-assistant.io/hassio/installation/) and did a fresh installation (it had a few false starts but eventually worked)
* Copied my backup back onto the Pi in the `/backup` folder
* Went to the web interface and hit the refresh icon on the snapshots page (upper-right)
* Selected my backup and chose restore
* Waited, rebooted, etc. 

After that I have a working Hassio again and was able to install Configurator. I feel like ideally there should be an easier way to do a full backup/restore that can actually address disk corruption issues like this (maybe caused by unreliable SD cards?), but this got things working.
